production.Dockerfile

```diff
@@ -64,6 +64,10 @@ RUN apk --update --no-cache --virtual .build-deps add \
     "git~=2" \
     "libffi-dev~=3.3" \
     "postgresql-dev~=13" \
+    "gcc" \
```

Comment:
@guidoiaquinti I suspect for this to work everywhere, I need to make changes elsewhere as well?

(Since you worked on this last, I was hoping you can help me out here?) Where all do I need to make changes to ensure things still work?

Reply:
Is this change required because we'll start using `scipy` instead of `numpy`? Also, I suspect `gcc` is only needed at `pip` compile time, can you move it to line `63:66`? 

As you probably already noticed our `Dockerfile` story is a bit a mess. We currently have

- `dev.Dockerfile`: for development, builds the `posthog/posthog:dev` Docker image and it's used by `docker-compose.dev.yaml`, ...
- `prod.Dockerfile`: for production, builds the `posthog/posthog:latest` Docker image and it's used by  `docker-compose.yaml`, ....

but we also have custom `Dockerfile`s in our `posthog-cloud` (the image built there is the one deployed on PostHog Cloud).

Reply:
Got it, thanks! Sounds like I'll have to update in all 3 places. Will make this a separate PR.

Reply:
And correct, we need stuff that numpy can't give us anymore, sadly

Reply:
👍 feel free to ping me on Slack if something is not clear, happy to help!